internal/stringutils/stringutils.go

```diff
@@ -0,0 +1,12 @@
+// Package stringutils provides utility functions for string operations.
+package stringutils
+
+// IsASCII checks if a string contains only ASCII characters.
+func IsASCII(s string) bool {
+	for i := 0; i < len(s); i++ {
+		if s[i] > 127 {
```

Comment:
Don't know if we want to mimic `isascii()` here, and in such case we could use the same logic (`(s[i] & ^0x7f) == 0`), however maybe we also want to actually want to mimic `isgraph` instead?